src/fastga_he/powertrain_builder/powertrain.py

```diff
@@ -544,6 +546,245 @@ def _get_connections(self):
+                        f"Having {input_count} inputs but expected {input_count_defined} for {comp}"
+                    )
+
+                elif int(output_count_defined) != int(output_count):
+                    raise OutputCountError(
+                        f"Having {output_count} outputs but expected {output_count_defined} "
+                        f"for {comp}"
+                    )
+
+        # Check one-to-one connections definition
+        for comp in self._components_name:
+            if comp not in propulsor_component and not any(
+                connection.get("target") == comp or comp in connection.get("target")
+                for connection in connections_list
+            ):
+                raise ComponentConnectionError(f"{comp} is missing as target!")
+
+            if comp not in energy_storage_component and not any(
+                connection.get("source") == comp or comp in connection.get("source")
+                for connection in connections_list
+            ):
+                raise ComponentConnectionError(f"{comp} is missing as source!")
+
+    def _categorize_connector_type_component(
+        self, one_to_one_component, connector_names, connector_options, connector_type
+    ):
+        """
+        This function categorizes the components in the connector component type class according
+        to their number of input and output connections, the generator and turbo_generator are
+        exceptions that are energy source component but categorized as connector for the
+        powertrain component registry. This only applies in the _check_connection function.
+        """
+        connector_variable = [
+            variable
+            for name in connector_names
+            for variable in self._components_connection_inputs + self._components_connection_outputs
+            if name in variable
+        ]
+
+        one_to_many_component = []
+        many_to_one_component = []
+        many_to_many_component = []
+        many_to_one_input_count = []
+        one_to_many_output_count = []
+        many_to_many_input_count = []
+        many_to_many_output_count = []
+
+        for name, options, type in zip(connector_names, connector_options, connector_type):
+            variable_list = [var for var in connector_variable if name in var]
+            defined_multi_connection_exists = options is not None and any(
+                key.startswith("number_of_") for key in options.keys()
+            )
+
+            if defined_multi_connection_exists:
+                # This is for the connectors having given input and output numbers from pt file
+                num_connections = [
+                    num
+                    for option, num in zip(options.keys(), options.values())
+                    if option.startswith("number_of_")
+                ]
+
+                # First check if there is any side having multiple connection
+                if any(num > 1 for num in num_connections):
```

Comment:
Do we need a for loop here ? Won't num_connections always be of size 2 ? So it is equivalent to:
```suggestion
                if num_connections[0] > 1 or num_connections[1] > 1:
```

I genuinely don't know which formulation is better